kong/reports.lua

```diff
@@ -57,7 +57,7 @@ local REQUEST_ROUTE_CACHE_HITS_KEY_NEG = REQUEST_COUNT_KEY .. ":" .. ROUTE_CACHE
 
 local _buffer = {}
 local _ping_infos = {}
-local _enabled = false
+local _enabled = kong and kong.configuration and kong.configuration.anonymous_reports or false
```

Comment:
Should we move initialization of this inside init_worker of this module?

Reply:
what about this: ^

Reply:
Aka change:
https://github.com/Kong/kong/blob/master/kong/reports.lua#L432-L434

the:
```
    if not _enabled then
      return
    end
```

to:

```
    _enabled = kong.configuration.anonymous_reports == true
    if not _enabled then
      return
    end
```

And remove the code from module level?

Reply:
@bungle  now that we have this change `_enabled = kong.configuration.anonymous_reports == true`, I'm trying to remove toggle from the code base. And FTI-4469 seems intend to respect anonymous_reports totally

Reply:
@bungle The calls of `toggle` have been removed in the latest commit. An extra work have to be done in EE is that distribution_constants should be treated correctly so that anonymous_reports could be respected.